src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs

```diff
@@ -198,5 +185,208 @@ public void SarifTransformerTests_ToVersionOne_OneRunWithLogicalLocations()
 
             VerifyCurrentToVersionOneTransformation(V2LogText, V1LogExpectedText);
         }
+
+        [Fact]
+        public void SarifTransformerTests_ToVersionOne_OneRunWithFiles()
+        {
+            const string V2LogText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-2.0.0"",
+  ""version"": ""2.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""invocations"": [
+        null
+      ],
+      ""files"": {
+        ""file:///home/list.txt"": {
+          ""length"": 43,
+          ""mimeType"": ""text/plain"",
+          ""contents"": {
+            ""text"": ""The quick brown fox jumps over the lazy dog"",
+            ""binary"": ""VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wcyBvdmVyIHRoZSBsYXp5IGRvZw==""
+          },
+          ""hashes"": [
+            {
+              ""value"": ""d7a8fbb307d7809469ca9abcb0082e4f8d5651e46d3cdb762d02d0bf37c9e592"",
+              ""algorithm"": ""sha-256""
+            }
+          ]
+        },
+        ""file:///home/buildAgent/bin/app.zip"": {
+          ""mimeType"": ""application/zip"",
+          ""contents"": {},
+          ""properties"": {
+            ""my_key"": ""some value""
+          }
+        },
+        ""file:///home/buildAgent/bin/app.zip#/docs/intro.docx"": {
+          ""fileLocation"": {
+            ""uri"": ""file:///docs/intro.docx""
+          },
+          ""parentKey"": ""file:///home/buildAgent/bin/app.zip"",
+          ""offset"": 17522,
+          ""length"": 4050,
+          ""mimeType"": ""application/vnd.openxmlformats-officedocument.wordprocessingml.document"",
+          ""contents"": {}
+        }
+      },
+      ""results"": []
+    }
+  ]
+}";
+            const string V1LogExpectedText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-1.0.0"",
+  ""version"": ""1.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""files"": {
+        ""file:///home/list.txt"": {
+          ""length"": 43,
+          ""mimeType"": ""text/plain"",
+          ""contents"": ""The quick brown fox jumps over the lazy dog"",
```

Comment:
So when I transform, should I decode it back to text and put the string in the Text property?

Reply:
Ah never mind, I'm getting myself all turned around.

Reply:
ok here is a question for the v1 -> v2 transformer: if contents is set, and the mime type is a text one, do I assume UTF8 and convert to a readable string that goes in fileContent.text?
byte[] data = Convert.FromBase64String(encodedString);
string decodedString = Encoding.UTF8.GetString(data);

Reply:
@EasyRhin0 So, to be clear, you want to put the value of v2's `file.contents.binary` here.

The hard question is what to do when the v2 `file.contents` has only a `text` property. I think the answer is:
- If you know `file.encoding`, and it's one that .NET supports, then encode the string with that encoding and populate `file.contents` with the MIME Base64 encoded version of the result.
- Otherwise, add a property `textContents` to the v1 `file`'s property bag.
- In any case, if v2 `file.encoding` is present, add `encoding` to the v1 `file`'s property bag.

And of course we need tests for all of these. I would use separate test methods to make the sample log files shorter and to clarify the intent of each test.